src/states/mod.rs

```diff
@@ -38,8 +40,30 @@ pub use self::{
 //        │   └────────────────┘ └─────────────┘
 //        │                        │
 //        │                        │
+//        │                        ▼
+//        │                      ┌──────────────────┐
+//        │                      │ EstablishingNode │
+//        │                      └──────────────────┘
+//        │                        │
+//        │                        │
 //        ▼                        ▼
 // ┌────────┐                    ┌──────┐
 // │ Client │                    │ Node │
 // └────────┘                    └──────┘
 //
+//
+// # Common traits
+//                           Bootstrapping
```

Comment:
`Bootstrapping` can become `Client` or `Node`, but the other states are all going to become only `Node`. I think it makes some sense to have this distinction. On the other hand I don't feel too strongly about it and having shorter names would be nice too. Your call.

Reply:
Hm, so I wonder: I can remove the `Node` suffix, but that would still make the `Client` and `Node` names inconsistent with the rest, as the rest are verbs in the present continuous tense (`Bootstrapping`, `Relocating`, `Proving`, ...), but `Client` and `Node` are nouns. I can maybe go the opposite way and rename `Node` and `Client` to `EstablishedNode` and `EstablishedClient`, but that would still leave `Bootstrapping` out as it can be both node and client. Not sure. Naming stuff is hard. Maybe best to leave this as is for now and maybe return to it in the future?

Reply:
you could switch `Bootstrapping` -> `BootstrappingPeer`. we normally go with `Peer` when common to Node & Client but not like it was enforced.